src/components/BooksList/BooksList.tsx

```diff
@@ -0,0 +1,50 @@
+import BookListItem from 'components/BookListItem/BookListItem';
+import React, { useEffect, useState } from 'react';
+import classes from './BooksList.module.scss';
+import data from 'data/books.json';
+import axios from 'axios';
+
+interface BooksListData {
+  id: number;
+  title: string;
+  author: string;
+  description: string;
+  imagePath: string;
+  numOfCopies: number;
+}
+
+const BooksList = () => {
+  useEffect(() => {
+    retriveBooks();
+  }, []);
+
+  const config = {
+    headers: {
+      Authorization:
+        'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJsaWJyYXJ5LmlvIiwiYXVkIjoibGlicmFyeSIsImlzcyI6ImlvLmxpYnJhcnkuYXBwIiwiZXhwIjoxNjY4MDAyNDMxLCJ1c2VySWQiOjEwMDAsImVtYWlsIjoiYWRtaW5AbnFvZGUuaW8iLCJ1c2VyUm9sZSI6IkFETUlOSVNUUkFUT1IifQ.XbaKJrfBP2SH3jQ7jdyDUPMw3su1HFSSo3E9zc-iYbM-er44Tva1I3WTaQURaTBFz8od4SaI6ZgjcfbqQMW7aA'
+    }
+  };
+
+  const [books, setBooks] = useState<BooksListData[]>([]);
+
+  const retriveBooks = async () => {
+    try {
+      const response = await axios.get(`http://localhost:8080/book`, config);
```

Comment:
This is a bad practice to pass link like this.
Please configure .env file to use connect to API from there.